### ISSUE TITLE
Removal of contract code (selfdestruct) was not reflected in hardhat 2.2.0

### ISSUE THREAD

**Issue by NIC619** [nomiclabs/hardhat#1415]
I upgraded `hardhat` from  `2.0.6` to `2.2.0` and ran into a problem where transaction that triggers a contract to `selfdestruct` succeeded (`receipt.status == 1`) but querying the dead contract's code still shows that the contract contains code (`ethers.provider.getCode(CONTRACT_ADDRESS)`).
Contract balance can be correctly updated however, but not the contract code.

To reproduce:
- contract code (modified from `Greeter` example)
```solidity
//SPDX-License-Identifier: Unlicense
pragma solidity ^0.7.0;

contract Greeter {

  constructor() payable {}

  function kill() public {
    selfdestruct(msg.sender);
  }

}
```

- test
```javascript
const { expect } = require("chai");

describe("Greeter", function() {
  it("Should successfully update contract state after selfdestruct", async function() {
    const Greeter = await ethers.getContractFactory("Greeter");
    const greeter = await Greeter.deploy({value: 1000});
    await greeter.deployed();

    expect(await ethers.provider.getBalance(greeter.address)).to.equal(1000)
    expect(await ethers.provider.getCode(greeter.address)).to.not.equal("0x")
    await greeter.kill();
    expect(await ethers.provider.getBalance(greeter.address)).to.equal(0)
    expect(await ethers.provider.getCode(greeter.address)).to.equal("0x")
  });
});
```

Then install different versions of `hardhat` and run the test. So far I've tried `2.0.6`, `2.0.10`, `2.1.0`, `2.1.2` and they all work fine. Only `2.2.0` fail.

**Comment by alcuadrado** [comment]
Hey @NIC619, thanks for reporting this bug and trying it with different versions! It was really helpful to understand what's going on. We are now working on a fix.

**Comment by fvictorio** [comment]
Thanks @NIC619, this should be fixed in `hardhat@2.2.1`.
